### ISSUE TITLE
Extension methods do not see each other when overloaded

### ISSUE THREAD

**Issue by anatoliykmetyuk** [lampepfl/dotty#7986]
## minimized code

```Scala
case class Project(name: String)
def (name: String) dependencies = ???
def (project: Project) dependencies = project.name.dependencies
```

<details>
<summary>Compilation output</summary>

```scala
-- [E008] Member Not Found Error: /Users/kmetiuk/Projects/scala3/pg/sandbox/iss1.scala:3:51
3 |def (project: Project) dependencies = project.name.dependencies
  |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  |                              value dependencies is not a member of String
1 error found
```
</details>

## expectation
The following compiles fine (`dependencies` -> `dependenciez`):

```scala
case class Project(name: String)
def (name: String) dependencies = ???
def (project: Project) dependenciez = project.name.dependencies
```

So the original code should also compile.

Possibly related: #7375, #7401.

**Comment by odersky** [comment]
It does not find the member because there is a cyclic reference, since `dependencies` does not have a result type.

**Comment by odersky** [comment]
Fixed by #10902
